cinema/views.py

```diff
@@ -1 +1,65 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Movie, Genre, Actor, MovieSession, CinemaHall
+from cinema.serializers import (
+    MovieSerializer,
+    MovieListSerializer,
+    MovieRetrieveSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    MovieSessionSerializer,
+    CinemaHallSerializer,
+    MovieSessionListSerializer,
+    MovieSessionRetrieveSerializer,
+)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+    def get_queryset(self):
+        queryset = Movie.objects.all()
+        if self.action == "list":
+            queryset = queryset.prefetch_related("genres", "actors")
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        if self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return MovieSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+    serializer_class = MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = MovieSession.objects.all()
+        if self.action == "list":
+            queryset = queryset.select_related("movie", "cinema_hall")
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionListSerializer
+        if self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+        return MovieSessionSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
```

Comment:
According to the checklist, you do not need to use `.all()` before `.prefetch_related()` and `.select_related()` methods. So you can remove `.all()` here.